### PR TITLE
fix(typescript): Add missing class functions

### PR DESCRIPTION
**Pull request by wolfy1339**

### CODE REVIEW

**Comment by wolfy1339** [comment]
This should be ready to go @gr2m

File: scripts/generate-types.js [hunk 1]
```diff
@@ -54,9 +56,16 @@ export namespace Webhooks {
 
     public on (event: 'error', callback: (event: Error) => void): void
     public on (event: '*' | string[], callback: (event: Webhooks.WebhookEvent<any>) => void): void
+    public on (event: '*' | string[], callback: (event: Webhooks.WebhookEvent<any>) => Promise<void>): void
     ${signatures.join('\n')}
 
     public sign (data: any): string
+    public verify (eventPayload: WebhookPayloadWithRepository, signature: string): boolean
```
Comment (gr2m) [inline scripts/generate-types.js]: Sorry, I’m not sure what `WebhookPayloadWithRepository` is? Not all event payload has a repository key, e.g. https://developer.github.com/v3/activity/events/types/#installationevent has `repositories`.

But if that works, can we use the same type for `sign()`? Both `sign()` and `verify()` accept the same payload type as first argument

File: scripts/generate-types.js [hunk 2]
```diff
@@ -54,9 +56,16 @@ export namespace Webhooks {
...
+    public verifyAndReceive (options: { id: string, name: string, payload: any, signature: string }): Promise<void>
+    public receive (options: { id: string, name: string, payload: any }): Promise<void>
+    public removeListener (event: string, callback: (event: Webhooks.WebhookEvent<any>) => void): void
+    public removeListener (event: string, callback: (event: Webhooks.WebhookEvent<any>) =>Promise<void>): void
```
Comment (gr2m) [inline scripts/generate-types.js]: `event` can be a `string` or an event of strings. In line 66, I think the return type is incorrect/

```diff
- (event: Webhooks.WebhookEvent<any>) => void)
+ (event: Webhooks.WebhookEvent<any>) => Promise<void>)
```

**Review by gr2m — COMMENTED** [review]
Thanks for working on this!

File: scripts/generate-types.js (hunk 1 above)
Comment (wolfy1339) [reply scripts/generate-types.js]: I had taken these from a PR of mine from Probot. I'll fix that. I didn't notice it

**Review by gr2m — APPROVED** [review]
Awesome, thank you!

**Pull request merged**